tools/parse_soundfont.py

```diff
@@ -31,24 +31,29 @@ def AudioHeap_InitAdsrDecayTable():
     release_values[253] = 1.0 / calculate_decay(0.5) / 60 * 0.7
     release_values[254] = 1.0 / calculate_decay(0.33) / 60 * 0.7
     release_values[255] = 1.0 / calculate_decay(0.25) / 60 * 0.7
-    
-
 
-def calculate_frequency(r, T, t):
-    exponent = (60 - (r - T - 0.01 * t)) / 24
-    f = 4 ** exponent
-    return f
+# Calculate tuning float for channel-based, or range-based, instruments (instruments)
+# Resulting final tuning value must result in "1 ÷ Float = Note Speed" (e.g. lower float = higher pitch)
+def calc_chanbased_tuning(r, s, c, hR, sR):
+    chanbased_tuning = 2 ** ((60 - (r - s - (0.01 * c))) / 12) / (hR / sR) # float * (sR / hR) would also work, just preference
+    return chanbased_tuning
 
-def calculate_inst_tuning(r, s, c, hR, sR):
-    exponent = calculate_frequency(r, s, c)
-    sample_rate_correction = sR / hR
-    tuning = exponent * sample_rate_correction
-    return tuning
+# Calculate tuning float for key-based instruments (drums and sound effects)
+# Resulting final tuning value must result in "1 * Float = Note Speed" (e.g. higher float = higher pitch)
+#
+# The tuning calculation already calculates as if the sample is assigned to 60. However, for SF2 parsing the distance from 60
+# must also be calculated and added as a secondary coarse tune: "Root Key - Key Range = Distance from 60"
+#
+# If you want the drum to play at the exact pitch of its root key, then the root key will be equal to the key range.
+# If you want to drum to play at a higher pitch than its root key, then the root key will be greater than the key range.
```

Comment:
the root key would be lower than the key range if you wanted it to be higher pitched...